dashboard/app/models/user.rb

```diff
@@ -486,10 +486,12 @@ def email_and_hashed_email_must_be_unique
     # skip the db lookup if we are already invalid
     return unless errors.blank?
 
-    if ((email.present? && (other_user = User.find_by_email_or_hashed_email(email))) ||
-        (hashed_email.present? && (other_user = User.find_by_hashed_email(hashed_email)))) &&
-        other_user != self
+    if email.present? && other_user != User.find_by_email_or_hashed_email(email) &&
+      self != other_user
       errors.add :email, I18n.t('errors.messages.taken')
+    elsif hashed_email.present? && other_user != User.find_by_hashed_email(hashed_email) &&
+      self != other_user
+      errors.add :hashed_email, I18n.t('errors.messages.taken')
```

Comment:
Note that this adds an error on `hashed_email`, not `email`. Is this desired? Not desired?

Though more accurate, it also might confuse users.